### PR TITLE
Saving data when they are changed

### PR DESCRIPTION
**Pull request by koushki**
Hey guys, I'm coming back :)

As mentioned in #1178 issue

### CODE REVIEW

**Comment by davidpersson** [comment]
Thanks for looking into the issue again. This looks good, especially as not much code is added. Before we can merge, we'll need more tests to make sure this doesn't break something. 

We will need tests for more edge cases.

**Comment by koushki** [comment]
Ok , would you like to me for add more testcases ?

**Comment by davidpersson** [comment]
@koushki Yeah, this would be great.

**Comment by koushki** [comment]
@davidpersson I added more integration test case but i'm not sure which kind of test case were you mean. 
If you think we can add more test case please give me more idea for write them .

**Comment by davidpersson** [comment]
Thanks for adding more cases. I was thinking about edge cases in particular. Off the top of my head:

* Saving all fields updated
* Saving no fields updated
* Saving just a few updated
* Saving non-existing field updated
* Saving using whitelist (all fields updated, none updated, some missing)

File: tests/integration/data/DatabaseTest.php [hunk 1]
```diff
@@ -483,6 +483,33 @@ public function testSubqueryWithMultipleRecords() {
 		));
 		$this->assertEqual(2, $results->count());
 	}
+
+	public function testUpdateWithMultiThread() {
```
Comment (davidpersson) [inline tests/integration/data/DatabaseTest.php]: We cannot test concurrency reliably/simply, so this test can be removed.

Comment (koushki) [inline tests/integration/data/DatabaseTest.php]: no it isn't parallelism, just the name is multiThread .

in this test case we have two different object that affect on each other.

Comment (davidpersson) [inline tests/integration/data/DatabaseTest.php]: Aaah, alrighty. Missed that.

**Comment by koushki** [comment]
@davidpersson Please check again , I added more test cases as you mentioned

**Comment by davidpersson** [comment]
Merged in 8b9a42701de3cf407e623a1f827510d11db97c6e. Sorry that the meta commit info got lost in that merge. Nonetheless thanks for the good work! Let's keep an eye on this.
